src/displayapp/screens/Adder.cpp

```diff
@@ -0,0 +1,340 @@
+}
+
+void Adder::UpdatePosition() {
+  unsigned int newHead = adderBody.front() + currentDirection;
+  Adder::MoveConsequence result = CheckMove();
+
+  switch (result) {
+    case Adder::MoveConsequence::DEATH:
+      ResetGame();
+      return;
+
+    case Adder::MoveConsequence::EAT:
+      adderBody.push_front(newHead);
+      CreateFood();
+      UpdateScore(adderBody.size() - 2);
+      break;
+
+    case Adder::MoveConsequence::MOVE:
+      adderBody.pop_back();
+      adderBody.push_front(newHead);
+      break;
+  }
+
+  field[adderBody.front()] = AdderField::BODY;
+  field[adderBody.back()] = AdderField::BLANK;
+}
+
+Adder::MoveConsequence Adder::CheckMove() const {
+  unsigned int newHead = adderBody.front() + currentDirection;
+  if (newHead >= fieldSize) {
+    return Adder::MoveConsequence::DEATH;
+  }
+
+  switch (field[newHead]) {
+    case AdderField::BLANK:
+      return Adder::MoveConsequence::MOVE;
+    case AdderField::FOOD:
+      return Adder::MoveConsequence::EAT;
+    default:
+      return Adder::MoveConsequence::DEATH;
+  }
+}
+
+void Adder::Refresh() {
+  if (!appReady) {
+    FullRedraw();
+    CreateFood();
+    vTaskDelay(1); // Required to let the OS draw the tile completely
+    UpdateScore(0);
+    vTaskDelay(1); // Required to let the OS draw the tile completely
+    appReady = true;
+  } else {
+    UpdatePosition();
+    UpdateSingleTile(adderBody.front() % fieldWidth, adderBody.front() / fieldWidth, LV_COLOR_YELLOW);
+    vTaskDelay(1); // Required to let the OS draw the tile completely
+    UpdateSingleTile(adderBody.back() % fieldWidth, adderBody.back() / fieldWidth, LV_COLOR_BLACK);
+    vTaskDelay(1); // Required to let the OS draw the tile completely
+  }
+}
+
+void Adder::FullRedraw() {
+  for (unsigned int x = 0; x < fieldWidth; ++x) {
+    for (unsigned int y = 0; y < fieldHeight; ++y) {
+      lv_color_t color;
+      switch (field[y * fieldWidth + x]) {
+        case AdderField::BODY:
+          color = LV_COLOR_YELLOW;
+          break;
+        case AdderField::SOLID:
+          color = LV_COLOR_WHITE;
+          break;
+        case AdderField::FOOD:
+          color = LV_COLOR_GREEN;
+          break;
+        default:
+          color = LV_COLOR_BLACK;
+          break;
+      }
+      UpdateSingleTile(x, y, color);
+      vTaskDelay(1); // Required to let the OS draw the tile completely
+    }
+  }
+}
+
+void Adder::UpdateSingleTile(unsigned int x, unsigned int y, lv_color_t color) {
+  std::fill(tileBuffer, tileBuffer + TileSize * TileSize, color);
+  lv_area_t area {.x1 = static_cast<lv_coord_t>(x * TileSize + fieldOffsetHorizontal),
+                  .y1 = static_cast<lv_coord_t>(y * TileSize + fieldOffsetVertical),
+                  .x2 = static_cast<lv_coord_t>(x * TileSize + fieldOffsetHorizontal + TileSize - 1),
+                  .y2 = static_cast<lv_coord_t>(y * TileSize + fieldOffsetVertical + TileSize - 1)};
+
+  lvgl.FlushDisplay(&area, tileBuffer);
```

Comment:
need to use `std::array::data()` to get a pointer for lvgl
```suggestion
  lvgl.FlushDisplay(&area, tileBuffer.data());
```